cpp/example/scaling.cc

```diff
@@ -0,0 +1,42 @@
+#include <cap/energy_storage_device.h>
+#include <cap/mp_values.h>
+#include <cap/default_inspector.h>
+#include <boost/property_tree/ptree.hpp>
+#include <boost/property_tree/info_parser.hpp>
+#include <boost/mpi/environment.hpp>
+#include <boost/mpi/timer.hpp>
+#include <iostream>
+
+int main(int argc, char *argv[])
+{
+  boost::mpi::environment env(argc, argv);
+  boost::mpi::communicator comm;
+  boost::mpi::timer timer;
+  if (comm.rank() == 0)
+    std::cout << "Number of processors: " << comm.size() << std::endl;
+
+  // Parse input file
+  boost::property_tree::ptree device_database;
+  boost::property_tree::info_parser::read_info("super_capacitor.info",
+                                               device_database);
+
+  std::shared_ptr<cap::EnergyStorageDevice> device =
+      cap::EnergyStorageDevice::build(device_database, comm);
```

Comment:
Yeah. We will need to talk about this because it leads to weird thing with the destructor which is called twice. I am not sure if it is because of the cast but it's very strange.